### PR TITLE
Support for interpolation (aten::upsample_nearest)

### PR DESCRIPTION
**Pull request by abhi-iyer**
# Description

Support for aten::upsample_nearest3d, aten::upsample_nearest2d, and aten::upsample_nearest1d

Also fixed a bug that forced input tensors to be at least 4 dimensional. 

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation and have regenerated the documentation (`make html` in docsrc)
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by narendasan** [comment]
@abhi-iyer I think you probably should rebase this branch on master to resolve the conflicts

File: core/conversion/converters/impl/interpolate.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#include <csignal>
+
+namespace trtorch {
+namespace core {
+namespace conversion {
+namespace converters {
+namespace impl {
+namespace {
+
+auto interpolate_registrations = RegisterNodeConversionPatterns()
```
Comment (narendasan) [inline core/conversion/converters/impl/interpolate.cpp:14]: Can you add the `TRTORCH_UNUSED` label here to suppress warnings on other platforms?

File: core/conversion/converters/impl/interpolate.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                TRTORCH_CHECK(resize_layer, "Unable to create interpolation (resizing) layer from node" << *n);
+
+                resize_layer->setOutputDimensions(util::toDims(out_shape));
+                resize_layer->setResizeMode(nvinfer1::ResizeMode::kNEAREST);
+                resize_layer->setName(util::node_info(n).c_str());
+
+                auto layer_output = ctx->AssociateValueAndTensor(n->outputs()[0], resize_layer->getOutput(0));
+                LOG_DEBUG("Output tensor shape: " << layer_output->getDimensions());
+            } else {
+                LOG_DEBUG("scale factor parameter not supported yet.");
```
Comment (narendasan) [inline core/conversion/converters/impl/interpolate.cpp:40]: I feel like this should either be a warning or a full error out, probably an error since the program shouldn't compile without this layer. Here is an example on how to throw an error

```
 TRTORCH_THROW_ERROR("Unable to retrieve all node inputs for node: " \
                                << util::node_info(n) << " (ctx.AddLayer)\nSpecifically failed to retrieve value for input: " \
                                << *input_node);
```

File: core/conversion/converters/impl/interpolate.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+namespace core {
+namespace conversion {
+namespace converters {
+namespace impl {
+namespace {
+
+auto interpolate_registrations = RegisterNodeConversionPatterns()
+    .pattern({
+        "aten::upsample_nearest1d(Tensor self, int[1] output_size, float? scales=None) -> (Tensor)",
+        [](ConversionCtx* ctx, const torch::jit::Node*n, args& args) -> bool {
```
Comment (narendasan) [inline core/conversion/converters/impl/interpolate.cpp:17]: nit: space between the `Node*` and `n`

File: core/conversion/converters/impl/interpolate.cpp [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                TRTORCH_CHECK(resize_layer, "Unable to create interpolation (resizing) layer from node" << *n);
+
+                resize_layer->setOutputDimensions(util::toDims(out_shape));
+                resize_layer->setResizeMode(nvinfer1::ResizeMode::kNEAREST);
+                resize_layer->setName(util::node_info(n).c_str());
+
+                auto layer_output = ctx->AssociateValueAndTensor(n->outputs()[0], resize_layer->getOutput(0));
+                LOG_DEBUG("Output tensor shape: " << layer_output->getDimensions());
+            } else {
+                LOG_DEBUG("scale factor parameters not supported yet.");
```
Comment (narendasan) [inline core/conversion/converters/impl/interpolate.cpp:70]: Same error comment

File: core/conversion/converters/impl/interpolate.cpp [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
```
Comment (narendasan) [inline core/conversion/converters/impl/interpolate.cpp:100]: Same error comment

File: test.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
+# import torch.nn as nn
```
Comment (narendasan) [inline test.py:1]: Let’s not commit this file and trace.ts into the repo. For future notice there is a git ignored directory you can create `<repo path>/experiments` where you can put these files so they don’t show up in your changes

File: core/conversion/converters/impl/interpolate.cpp [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+                TRTORCH_CHECK(resize_layer, "Unable to create interpolation (resizing) layer from node" << *n);
+
+                resize_layer->setOutputDimensions(util::toDims(out_shape));
+                resize_layer->setResizeMode(nvinfer1::ResizeMode::kNEAREST);
+                resize_layer->setName(util::node_info(n).c_str());
+
+                auto layer_output = ctx->AssociateValueAndTensor(n->outputs()[0], resize_layer->getOutput(0));
+                LOG_DEBUG("Output tensor shape: " << layer_output->getDimensions());
+            } else {
+                TRTORCH_THROW_ERROR("Unable to retrieve all node inputs for node: " << util::node_info(n) << "\nScale factor parameter not supported yet");
```
[the remark this answers is not in the document]
Comment (abhi-iyer) [reply core/conversion/converters/impl/interpolate.cpp]: done
